subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,486 @@
+                                                      season=candidate.get('season'), year=candidate.get('year'),
+                                                      no_downloads=no_downloads, rating=rating, featured=featured,
+                                                      multiple_episodes=multiple_episodes, timestamp=timestamp)
+
+                        logger.debug('Found subtitle %s', subtitle)
+                        subtitles.append(subtitle)
+
+                next_page_link = soup.find('a', attrs={'class': 'load_more'}, text='carregar mais')
+                page_url = self.server_url + next_page_link['href'] if next_page_link else None
+
+        # High quality subtitles should have higher precedence when their scores are equal.
+        subtitles.sort(key=lambda s: (s.featured, s.no_downloads, s.rating, s.multiple_episodes), reverse=True)
+
+        return subtitles
+
+    def list_subtitles(self, video, languages):
+        """
+        Returns a list of subtitles for the defined video and requested languages
+
+        :param video:
+        :param languages: the requested languages
+        :return: a list of subtitles for the requested video and languages
+        :rtype : ``list`` of ``LegendasTvSubtitles``
+        """
+        season = episode = None
+        if isinstance(video, Episode):
+            title = video.series
+            season = video.season
+            episode = video.episode
+        else:
+            title = video.title
+        year = video.year
+
+        return [s for l in languages for s in self.query(l, title, season=season, episode=episode, year=year)]
+
+    def get_subtitle_names(self, content):
+        """
+        Returns all subtitle names for a specific subtitle_id. Only subtitle names are returned.
+
+        :param content: the downloaded binary content (rar/zip)
+        :return: list of subtitle names
+        :rtype: ``list`` of ``string``
+        """
+        return self._uncompress(
+            content,
+            lambda cf: [f for f in cf.namelist()
+                        if 'legendas.tv' not in f.lower() and f.lower().endswith(SUBTITLE_EXTENSIONS)])
+
+    def extract_subtitle(self, content, subtitle_name):
+        """
+        Extract the subtitle content from the compressed file. The file is downloaded, the subtitle_name is uncompressed
+        and its contents is returned.
+
+        :param content: the downloaded binary content (rar/zip)
+        :param subtitle_name: the filename to be extracted
+        :return: the subtitle content
+        :rtype : ``string``
+        """
+        return self._uncompress(content, lambda cf, name: fix_line_ending(cf.read(name)), subtitle_name)
+
+    def _uncompress(self, content, function, *args, **kwargs):
+        # Download content might be a rar file (most common) or a zip.
+        # Unfortunately, rarfile module only works with files (no in-memory streams)
```

Comment:
You can check the history:
https://github.com/markokr/rarfile/commits/master/rarfile.py

This was introduced on 20.11.2013 for version 2.7 by this commit: https://github.com/markokr/rarfile/commit/3ed04e372f472464634625d234efb8e56677d5ad#diff-5689b095d6dc734b474f99921552f98c

I could get rid of the temporary files, but the rarfile dependency should be changed from rarfile>=2.4 to rarfile>=2.7